src/range.rs

```diff
@@ -363,29 +467,61 @@ fn interval_to_string<V: Version>((start, maybe_end): &Interval<V>) -> String {
 #[cfg(test)]
 pub mod tests {
     use proptest::prelude::*;
-
-    use crate::version::NumberVersion;
+    use proptest::test_runner::TestRng;
 
     use super::*;
 
-    pub fn strategy() -> impl Strategy<Value = Range<NumberVersion>> {
-        prop::collection::vec(any::<u32>(), 0..10).prop_map(|mut vec| {
-            vec.sort_unstable();
-            vec.dedup();
-            let mut pair_iter = vec.chunks_exact(2);
-            let mut segments = SmallVec::empty();
-            while let Some([v1, v2]) = pair_iter.next() {
-                segments.push((NumberVersion(*v1), Some(NumberVersion(*v2))));
-            }
-            if let [v] = pair_iter.remainder() {
-                segments.push((NumberVersion(*v), None));
-            }
-            Range { segments }
-        })
+    pub fn strategy() -> impl Strategy<Value = Range<u32>> {
+        prop::collection::vec(any::<u32>(), 0..10)
```

Comment:
`prop_perturb` does not shrink well. I think we can avoid it by generating 2 `bool`s for whether it is unbounded below and above, and `(u32, bool)` for whether that element is exclusive.

Reply:
I rewrote the strategy. Is this more along the lines you were thinking?

Reply:
Yes, thank you!